value/src/it/functional/src/test/java/com/google/auto/value/GradleTest.java

```diff
@@ -0,0 +1,170 @@
+
+  @Test
+  public void basic() throws IOException {
+    String autoValueVersion = System.getProperty("autoValueVersion");
+    assertThat(autoValueVersion).isNotNull();
+    String localRepository = System.getProperty("localRepository");
+    assertThat(localRepository).isNotNull();
+
+    // Set up the fake Gradle project.
+    String buildGradleText = expandSystemProperties(BUILD_GRADLE_TEXT);
+    writeFile(fakeProject.newFile("build.gradle"), buildGradleText);
+    File srcDir = fakeProject.newFolder("src", "main", "java", "com", "example");
+    writeFile(new File(srcDir, "Foo.java"), FOO_TEXT);
+
+    // Build it the first time.
+    BuildResult result1 = buildFakeProject();
+    assertThat(result1.getOutput()).contains(
+        "Full recompilation is required because no incremental change information is available");
+    File output = new File(
+        fakeProject.getRoot(), "build/classes/java/main/com/example/AutoValue_Foo.class");
+    assertThat(output.exists()).isTrue();
+
+    // Add a source file to the project.
+    String barText = FOO_TEXT.replace("Foo", "Bar");
+    File barFile = new File(srcDir, "Bar.java");
+    writeFile(new File(srcDir, "Bar.java"), barText);
+
+    // Build it a second time.
+    BuildResult result2 = buildFakeProject();
+    assertThat(result2.getOutput()).doesNotContain("Full recompilation is required");
+
+    // Remove the second source file and build a third time. If incremental annotation processing
+    // is not working, this will produce a message like this:
+    //   Full recompilation is required because com.google.auto.value.processor.AutoValueProcessor
+    //   is not incremental
+    assertThat(barFile.delete()).isTrue();
+    BuildResult result3 = buildFakeProject();
+    assertThat(result3.getOutput()).doesNotContain("Full recompilation is required");
+  }
+
+  private BuildResult buildFakeProject() throws IOException {
+    GradleRunner runner = GradleRunner.create()
+        .withProjectDir(fakeProject.getRoot())
+        .withArguments("--info", "compileJava");
+    if (GRADLE_INSTALLATION.isPresent()) {
+      runner.withGradleInstallation(GRADLE_INSTALLATION.get());
+    } else {
+      runner.withGradleVersion(GradleVersion.current().getVersion());
+    }
+    return runner.build();
+  }
+
+  private static Optional<File> getGradleInstallation() {
+    String gradleHome = System.getenv("GRADLE_HOME");
+    if (gradleHome != null) {
+      File gradleHomeFile = new File(gradleHome);
+      if (gradleHomeFile.isDirectory()) {
+        return Optional.of(new File(gradleHome));
+      }
+    }
+    Path installationPath;
+    try {
+      Path gradleExecutable = Paths.get("/usr/bin/gradle");
+      Path gradleLink = Files.readSymbolicLink(gradleExecutable);
+      if (!gradleLink.isAbsolute()) {
+        gradleLink = gradleExecutable.getParent().resolve(gradleLink);
+      }
+      if (!gradleLink.toString().endsWith("/bin/gradle")) {
```

Comment:
This could be `!gradleLink.endsWith("bin/gradle")`. Unlike a string comparison, that wouldn't be true if the string form ended in `"somethingbin/gradle"`.